### ISSUE TITLE
TotalLoss condition for checkpoint save.

### ISSUE THREAD

**Issue by turowicz** [tensorflow/models#7717]
I'm on TF 1.x and my question is if there is a possibility to make the checkpoint saver only create checkpoints when the TotalLoss is below a certain threshold? 

I'm getting a large deviation on the TotalLoss and I would like to make sure the checkpoints I get are all usable as models. 

Checkpoints are saved often enough to be sure I can resume the training process.

**Comment by turowicz** [comment]
@saberkun perhaps you can help?

**Comment by saberkun** [comment]
You probably need to override estimator saverhook. I don't have an example to share.
If you use TF 2.x, if you directly manage checkpointing and the loss tensor is directly accessible through numpy() method.

**Comment by turowicz** [comment]
Thanks for the info @saberkun - unfortunately we aren't able to upgrade to TF2.x as the object-detector models from the ZOO are on 1.12.

Maybe looking at it from the other way - is it possible to check what the totalloss of a checkpoint file is? That way I could remove the ones that aren't interesting to me.

**Comment by saberkun** [comment]
is it possible to assign total loss to a variable so that the variable will be write into checkpoint. You can check the variable then.

**Comment by turowicz** [comment]
Any way examples of writing variables to checkpoint?
